commands/audit/audit.go

```diff
@@ -256,20 +259,72 @@ func isEntitledForJas(xrayManager *xray.XrayServicesManager, auditParams *AuditP
 }
 
-func downloadAnalyzerManagerAndRunScanners(auditParallelRunner *utils.SecurityParallelRunner, scanResults *utils.Results,
+func downloadAnalyzerManagerAndRunScanners(auditParallelRunner *utils.SecurityParallelRunner, scanResults *results.SecurityCommandResults,
 	serverDetails *config.ServerDetails, auditParams *AuditParams, scanner *jas.JasScanner, jfrogAppsConfig *jfrogappsconfig.JFrogAppsConfig, threadId int) (err error) {
 	defer func() {
 		auditParallelRunner.JasWg.Done()
 	}()
 	if err = jas.DownloadAnalyzerManagerIfNeeded(threadId); err != nil {
 		return fmt.Errorf("%s failed to download analyzer manager: %s", clientutils.GetLogMsgPrefix(threadId, false), err.Error())
 	}
-	scanner, err = jas.CreateJasScanner(scanner, jfrogAppsConfig, serverDetails, jas.GetAnalyzerManagerXscEnvVars(auditParams.commonGraphScanParams.MultiScanId, scanResults.GetScaScannedTechnologies()...), auditParams.Exclusions()...)
+	auditParallelRunner.ResultsMu.Lock()
+	scanner, err = jas.CreateJasScanner(scanner, serverDetails, jas.GetAnalyzerManagerXscEnvVars(auditParams.commonGraphScanParams.MultiScanId, scanResults.GetTechnologies()...), auditParams.Exclusions()...)
+	auditParallelRunner.ResultsMu.Unlock()
 	if err != nil {
 		return fmt.Errorf("failed to create jas scanner: %s", err.Error())
 	}
-	if err = runner.AddJasScannersTasks(auditParallelRunner, scanResults, auditParams.DirectDependencies(), serverDetails, auditParams.thirdPartyApplicabilityScan, scanner, applicability.ApplicabilityScannerType, secrets.SecretsScannerType, auditParallelRunner.AddErrorToChan, auditParams.ScansToPerform()); err != nil {
-		return fmt.Errorf("%s failed to run JAS scanners: %s", clientutils.GetLogMsgPrefix(threadId, false), err.Error())
+	// Run JAS scanners for each scan target
+	for _, scan := range scanResults.Targets {
+		module := jas.GetModule(scan.Target, jfrogAppsConfig)
+		if module == nil {
+			scan.AddError(fmt.Errorf("can't find module for path %s", scan.Target))
+			continue
+		}
+		if err = runner.AddJasScannersTasks(auditParallelRunner, *module, scan, auditParams.DirectDependencies(), serverDetails, auditParams.thirdPartyApplicabilityScan, scanner, applicability.ApplicabilityScannerType, secrets.SecretsScannerType, auditParams.ScansToPerform()); err != nil {
+			return fmt.Errorf("%s failed to run JAS scanners: %s", clientutils.GetLogMsgPrefix(threadId, false), err.Error())
+		}
+	}
+	return
+}
+
+func initCmdResults(entitledForJas bool, params *AuditParams) (cmdResults *results.SecurityCommandResults) {
+	cmdResults = results.NewCommandResults(params.xrayVersion, entitledForJas).SetMultiScanId(params.commonGraphScanParams.MultiScanId)
+	detectScanTargets(cmdResults, params)
+	scanInfo, err := coreutils.GetJsonIndent(cmdResults)
+	if err != nil {
+		return
 	}
+	log.Info(fmt.Sprintf("Preforming %d scans:\n%s", len(cmdResults.Targets), scanInfo))
 	return
 }
+
+func detectScanTargets(cmdResults *results.SecurityCommandResults, params *AuditParams) {
```

Comment:
I'd add some documentation for this function/ change name. scan 'targets' is not a clear enough name for someone who is not familiar with the flow I think